File: 2864-Maximum-Odd-Binary-Number/go/main.go
func maximumOddBinaryNumber(s string) string {
    sthStr := s
    lenS := len(sthStr)
    var zeros []string
    var ones []string

    for _, char := range sthStr {
        if char == '1' {
            ones = append(ones, "1")
        } else {
            zeros = append(zeros, "0")
        }
    }

    switch len(ones) {
    case 0:
        return strings.Repeat("0", lenS)
    case 1:
        return strings.Repeat("0", lenS-1) + "1"
    default:
        ones = ones[:len(ones)-1]
        return strings.Join(ones, "") + strings.Join(zeros, "") + "1"
    }
}

File: 2864-Maximum-Odd-Binary-Number/ts/main.js
/**
 * @param {string} s
 * @return {string}
 */
var maximumOddBinaryNumber = function(s) {
    let sth_str = s;
    let len = sth_str.length;
    let zeros = [];
    let ones = [];

    for (let i = 0; i < sth_str.length; i++) {
        if (sth_str[i] == '1') {
            ones.push('1');
        } else {
            zeros.push('0');
        }
    }

    let res;
    switch (ones.length) {
        case 0:
            res = new Array(len).fill("0").join('');
            return res;
        case 1:
            res = new Array(len - 1).fill("0").join('') + "1";
            return res;
        default:
            ones.pop();
            res = ones.join('') + zeros.join('') + "1";
            return res;
    }
};
